File: modules/java/android_test/src/org/opencv/test/core/KeyPointTest.java
package org.opencv.test.core;

import org.opencv.core.Point;
import org.opencv.core.KeyPoint;
import org.opencv.test.OpenCVTestCase;

public class KeyPointTest extends OpenCVTestCase {

    private float angle;
    private int classId;
    private KeyPoint keyPoint;
    private int octave;
    private float response;
    private float size;
    private float x;
    private float y;

    @Override
    protected void setUp() throws Exception {
        super.setUp();

        keyPoint = null;
        x = 1.0f;
        y = 2.0f;
        size = 3.0f;
        angle = 30.0f;
        response = 2.0f;
        octave = 1;
        classId = 1;
    }

    public void testKeyPoint() {
        keyPoint = new KeyPoint();
        assertPointEquals(new Point(0, 0), keyPoint.pt, EPS);
    }

    public void testKeyPointFloatFloatFloat() {
        keyPoint = new KeyPoint(x, y, size);
        assertPointEquals(new Point(1, 2), keyPoint.pt, EPS);
    }

    public void testKeyPointFloatFloatFloatFloat() {
        keyPoint = new KeyPoint(x, y, size, 10.0f);
        assertEquals(10.0f, keyPoint.angle);
    }

    public void testKeyPointFloatFloatFloatFloatFloat() {
        keyPoint = new KeyPoint(x, y, size, 1.0f, 1.0f);
        assertEquals(1.0f, keyPoint.response);
    }

    public void testKeyPointFloatFloatFloatFloatFloatInt() {
        keyPoint = new KeyPoint(x, y, size, 1.0f, 1.0f, 1);
        assertEquals(1, keyPoint.octave);
    }

    public void testKeyPointFloatFloatFloatFloatFloatIntInt() {
        keyPoint = new KeyPoint(x, y, size, 1.0f, 1.0f, 1, 1);
        assertEquals(1, keyPoint.class_id);
    }

    public void testToString() {
        keyPoint = new KeyPoint(x, y, size, angle, response, octave, classId);

        String actual = keyPoint.toString();

        String expected = "KeyPoint [pt={1.0, 2.0}, size=3.0, angle=30.0, response=2.0, octave=1, class_id=1]";
        assertEquals(expected, actual);
    }

}

File: modules/core/misc/objc/test/KeyPointTest.swift
//
//  KeyPointTest.swift
//
//  Created by Giles Payne on 2020/01/31.
//

import XCTest
import OpenCV

class KeyPointTest: OpenCVTestCase {

    let angle:Float = 30
    let classId:Int32 = 1
    let octave:Int32 = 1
    let response:Float = 2.0
    let size:Float = 3.0
    let x:Float = 1.0
    let y:Float = 2.0

    func testKeyPoint() {
        let keyPoint = KeyPoint()
        assertPoint2fEquals(Point2f(x: 0, y: 0), keyPoint.pt, OpenCVTestCase.FEPS)
    }

    func testKeyPointFloatFloatFloat() {
        let keyPoint = KeyPoint(x: x, y: y, size: size)
        assertPoint2fEquals(Point2f(x: 1, y: 2), keyPoint.pt, OpenCVTestCase.FEPS)
    }

    func testKeyPointFloatFloatFloatFloat() {
        let keyPoint = KeyPoint(x: x, y: y, size: size, angle: 10.0)
        XCTAssertEqual(10.0, keyPoint.angle);
    }

    func testKeyPointFloatFloatFloatFloatFloat() {
        let keyPoint = KeyPoint(x: x, y: y, size: size, angle: 1.0, response: 1.0)
        XCTAssertEqual(1.0, keyPoint.response)
    }

    func testKeyPointFloatFloatFloatFloatFloatInt() {
        let keyPoint = KeyPoint(x: x, y: y, size: size, angle: 1.0, response: 1.0, octave: 1)
        XCTAssertEqual(1, keyPoint.octave)
    }

    func testKeyPointFloatFloatFloatFloatFloatIntInt() {
        let keyPoint = KeyPoint(x: x, y: y, size: size, angle: 1.0, response: 1.0, octave: 1, classId: 1)
        XCTAssertEqual(1, keyPoint.classId)
    }

    func testToString() {
        let keyPoint = KeyPoint(x: x, y: y, size: size, angle: angle, response: response, octave: octave, classId: classId)

        let actual = "\(keyPoint)"

        let expected = "KeyPoint { pt: Point2f {1.000000,2.000000}, size: 3.000000, angle: 30.000000, response: 2.000000, octave: 1, classId: 1}"
        XCTAssertEqual(expected, actual)
    }

}
